react/features/token-data/TokenData.js

```diff
@@ -0,0 +1,92 @@
+/* @flow */
+
+import jwtDecode from 'jwt-decode';
+
+/**
+ * Represent the data parsed from the JWT token
+ */
+export default class TokenData {
+
+    /**
+     * Object representing callee data.
+     */
+    callee: Object;
+
+    /**
+     * Object representing caller data.
+     */
+    caller: Object;
+
+    /**
+     * Contains group.
+     */
+    group: string;
+
+    /**
+     * Indicates whether user is guest.
+     */
+    isGuest: boolean;
+
+    /**
+     * Raw and encoded JWT token.
+     */
+    jwt: string;
+
+    /**
+     * Payload object from parsed JWT token.
+     */
+    payload: Object;
+
+    /**
+     * Contains server.
+     */
+    server: string;
+
+    /**
+     * Creates instance of token data.
+     *
+     * @param {Object} config - Application config.
+     * @param {string} jwt - The JWT token.
+     */
+    constructor(config: Object, jwt: string) {
+        this.isGuest = config.enableUserRolesBasedOnToken !== true;
+
+        if (!jwt) {
+            return;
+        }
+
+        this.jwt = jwt;
+
+        this._decode();
+    }
+
+    /**
+     * Decodes the JWT token and sets the decoded data to properties.
+     *
+     * @returns {void}
+     */
+    _decode() {
+        this.payload = jwtDecode(this.jwt);
+
+        if (!this.payload) {
+            return;
+        }
+
+        if (!this.payload.context) {
+            return;
+        }
+
+        this.server = this.payload.context.server;
+        this.group = this.payload.context.group;
+        const callerData = this.payload.context.user;
+        const calleeData = this.payload.context.callee;
+
+        if (callerData) {
+            this.caller = callerData;
```

Comment:
I don't like leaving things unassigned. Users of this class are going to find caller and callee undefined anyway, so let's always assign them.